app/src/main/java/com/epfl/drawyourpath/database/FireDatabase.kt

```diff
@@ -237,19 +237,65 @@ class FireDatabase : Database() {
 
     override fun removeUserFromFriendlist(userId: String): CompletableFuture<Unit> {
         val currentUserId = getUserId()
-        if(currentUserId==null){
+        if (currentUserId == null) {
             val future = CompletableFuture<Unit>()
             future.completeExceptionally(Exception("Any user is logged !"))
             return future
-        }else{
+        } else {
             //remove the userId from the friendlist of the current user
-            return removeUserIdToFriendList(currentUserId, userId).thenApply{
+            return removeUserIdToFriendList(currentUserId, userId).thenApply {
                 //remove the current userId to the friendlist of the user with userId
                 removeUserIdToFriendList(userId, currentUserId)
             }
         }
     }
 
+    override fun addRunToHistory(run: Run): CompletableFuture<Unit> {
+        //check if user is correctly logged
+        val currentUserId = getUserId()
+        val future = CompletableFuture<Unit>()
+        if (currentUserId == null) {
+            future.completeExceptionally(Exception("Any user is logged !"))
+            return future
+        }
+
+        //create the field for the new path, the key is the start time
+        val newRun = HashMap<String, Any>()
+        newRun.put(run.getStartTime().toString(), run)
+        //update the runs history on the database
+        accessUserAccountFile(currentUserId).child(runsHistoryFile)
+            .updateChildren(newRun as Map<String, Any>)
+            .addOnSuccessListener { future.complete(Unit) }
+            .addOnFailureListener { future.completeExceptionally(it) }
+        return future
+
+    }
+
+    override fun removeRunFromHistory(run: Run): CompletableFuture<Unit> {
+        //check if user is correctly logged
+        val currentUserId = getUserId()
+        val future = CompletableFuture<Unit>()
+        if (currentUserId == null) {
+            future.completeExceptionally(Exception("Any user is logged !"))
+            return future
+        }
+
+        //obtain the previous history
+        accessUserAccountFile(currentUserId).child(runsHistoryFile).get()
```

Comment:
Reading the entire history to remove a run seems overkill. As we have the key of the run, couldn't we just delete directly?